pkg/storage/ocs.go

```diff
@@ -0,0 +1,415 @@
+	}
+
+	return builder.Object.Spec.ManageNodes, nil
+}
+
+// GetManagedResources fetches storageCluster managedResources value.
+func (builder *StorageClusterBuilder) GetManagedResources() (*ocsoperatorv1.ManagedResourcesSpec, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof("Getting storageCluster %s in namespace %s managedResources configuration",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("storageCluster object %s does not exist in namespace %s",
+			builder.Definition.Name, builder.Definition.Namespace)
+	}
+
+	return &builder.Object.Spec.ManagedResources, nil
+}
+
+// GetMonDataDirHostPath fetches storageCluster monDataDirHostPath value.
+func (builder *StorageClusterBuilder) GetMonDataDirHostPath() (string, error) {
+	if valid, err := builder.validate(); !valid {
+		return "", err
+	}
+
+	glog.V(100).Infof("Getting storageCluster %s in namespace %s monDataDirHostPath configuration",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	if !builder.Exists() {
+		return "", fmt.Errorf("storageCluster object %s does not exist in namespace %s",
+			builder.Definition.Name, builder.Definition.Namespace)
+	}
+
+	return builder.Object.Spec.MonDataDirHostPath, nil
+}
+
+// GetMultiCloudGateway fetches storageCluster multiCloudGateway value.
+func (builder *StorageClusterBuilder) GetMultiCloudGateway() (*ocsoperatorv1.MultiCloudGatewaySpec, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof("Getting storageCluster %s in namespace %s multiCloudGateway configuration",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("storageCluster object %s does not exist in namespace %s",
+			builder.Definition.Name, builder.Definition.Namespace)
+	}
+
+	return builder.Object.Spec.MultiCloudGateway, nil
+}
+
+// GetStorageDeviceSets fetches storageCluster storageDeviceSets value.
+func (builder *StorageClusterBuilder) GetStorageDeviceSets() ([]ocsoperatorv1.StorageDeviceSet, error) {
+	if valid, err := builder.validate(); !valid {
+		return nil, err
+	}
+
+	glog.V(100).Infof("Getting storageCluster %s in namespace %s storageDeviceSets configuration",
+		builder.Definition.Name, builder.Definition.Namespace)
+
+	if !builder.Exists() {
+		return nil, fmt.Errorf("storageCluster object %s does not exist in namespace %s",
+			builder.Definition.Name, builder.Definition.Namespace)
+	}
+
+	return builder.Object.Spec.StorageDeviceSets, nil
+}
+
+// WithManageNodes sets the storageCluster's managedNodes value.
+func (builder *StorageClusterBuilder) WithManageNodes(expectedManagedNodesValue bool) *StorageClusterBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof(
+		"Setting storageCluster %s in namespace %s with managedNodes value: %v",
```

Comment:
Please replace %v formatter to %t. The expectedManagedNodesValue argument is boolean